### PR TITLE
Implementation of frontend app for the Element integration

### PR DESCRIPTION
**Pull request by LarsLuz**
Includes app for Phoenix client and file-picker widget for usage as Matrix widget

### CODE REVIEW

File: element-messenger/src/ElementFrame.vue [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    <div v-if="!configuration.element || !configuration.element.server">
+      {{$gettext('Unfortunately Element was not configured correctly.')}}
+    </div>
+  </div>
+</template>
+<script>
+export default {
+  name: 'ElementFrame',
+  data: () => ({
+    configuration: window.phoenixConfig || []
```
Comment (LukasHirt) [inline element-messenger/src/ElementFrame.vue:15]: Access pls config via the store. We want to get rid of the `window.phoenixConfig` object.

**Review by LukasHirt — COMMENTED** [review]
Left one comment regarding `configuration`. I have however one question regarding the structure of the code. Is the file picker widget supposed to be used outside of Phoenix? Is that why it is wrapped in its own Vue instance?

**Comment by LarsLuz** [comment]
> Left one comment regarding `configuration`. I have however one question regarding the structure of the code. Is the file picker widget supposed to be used outside of Phoenix? Is that why it is wrapped in its own Vue instance?

Hello, sorry that I did not inform you about this particular situation. I've moved the file picker outside the main app project because its vue dependencies did clash with the version used by Phoenix. (I've installed it as dependency via yarn)

See the following error message:
```
Vue packages version mismatch:

- vue@2.6.12 (/home/lars.luz/Projects/oc/phoenix/apps/element-messenger/node_modules/vue/dist/vue.runtime.common.js)
- vue-template-compiler@2.6.11 (/home/lars.luz/Projects/oc/phoenix/apps/element-messenger/node_modules/vue-template-compiler/package.json)

This may cause things to work incorrectly. Make sure to use the same version for both.
If you are using vue-loader@>=10.0, simply update vue-template-compiler.
If you are using vue-loader@<10.0 or vueify, re-installing vue-loader/vueify should bump vue-template-compiler to the latest.
```
Unfortunately I did not have the time to troubleshoot this and I wasn't sure if I even could fix it without upgrading the dependencies of either the file-picker or Phoenix. Therefore the picker resides on it's own index.html file instead of a fullscreen component within the Phoenix vue component.

**Comment by LukasHirt** [comment]
@LarsLuz I see. I will debug the issue with versions in the next sprint (next two weeks). Don't want to create a blocker from that for your PR. Would be nice to take care of it later when it's working properly though.

One more small detail - you have both `yarn.lock` and `package-lock.json`. Please, stick to only one of them and add the other into `.gitignore`. We're using `yarn.lock` in all other repos - would be great to keep that one here as well.

**Pull request merged**